src/libs/actions/Task.js

```diff
@@ -578,40 +629,65 @@ function setShareDestinationValue(shareDestination) {
+    Onyx.merge(ONYXKEYS.TASK, {assigneeChatReport: chatReport});
+}
+
 /**
  * Sets the assignee value for the task and checks for an existing chat with the assignee
  * If there is no existing chat, it creates an optimistic chat report
  * It also sets the shareDestination as that chat report if a share destination isn't already set
- * @param {string} assignee
+ * @param {string} assigneeEmail
  * @param {Number} assigneeAccountID
  * @param {string} shareDestination
  * @param {boolean} isCurrentUser
  */
 
-function setAssigneeValue(assignee, assigneeAccountID, shareDestination, isCurrentUser = false) {
-    let newAssigneeAccountID = Number(assigneeAccountID);
+function setAssigneeValue(assigneeEmail, assigneeAccountID, shareDestination, isCurrentUser = false) {
+    let chatReport;
 
-    // Generate optimistic accountID if this is a brand new user account that hasn't been created yet
-    if (!newAssigneeAccountID) {
-        newAssigneeAccountID = UserUtils.generateAccountID(assignee);
-    }
     if (!isCurrentUser) {
-        let newChat = {};
-        const chat = ReportUtils.getChatByParticipants([newAssigneeAccountID]);
-        if (!chat) {
-            newChat = ReportUtils.buildOptimisticChatReport([newAssigneeAccountID]);
+        chatReport = ReportUtils.getChatByParticipantsByLoginList([assigneeEmail]) || ReportUtils.getChatByParticipants([assigneeAccountID]);
+        if (!chatReport) {
+            chatReport = ReportUtils.buildOptimisticChatReport([assigneeAccountID]);
+            chatReport.isOptimisticReport = true;
+
+            // When assigning a task to a new user, by default we share the task in their DM
+            // However, the DM doesn't exist yet - and will be created optimistically once the task is created
+            // We don't want to show the new DM yet, because if you select an assignee and then change the assignee, the previous DM will still be shown
+            // So here, we create it optimistically to share it with the assignee, but we have to hide it until the task is created
+            chatReport.isHidden = true;
+            Onyx.set(`${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`, chatReport);
+
+            // If this is an optimistic report, we likely don't have their personal details yet so we set it here optimistically as well
+            const optimisticPersonalDetailsListAction = {
+                accountID: assigneeAccountID,
+                avatar: lodashGet(allPersonalDetails, [assigneeAccountID, 'avatar'], UserUtils.getDefaultAvatarURL(assigneeAccountID)),
+                displayName: lodashGet(allPersonalDetails, [assigneeAccountID, 'displayName'], assigneeEmail),
+                login: assigneeEmail,
+            };
+            Onyx.merge(ONYXKEYS.PERSONAL_DETAILS_LIST, {[assigneeAccountID]: optimisticPersonalDetailsListAction});
         }
-        const reportID = chat ? chat.reportID : newChat.reportID;
 
+        setAssigneeChatReport(chatReport);
```

Comment:
This line introduces a bug [here](https://github.com/Expensify/App/issues/38439) where the user chose a brand new assignee (whom they've never chatted with before), and then chose another old assignee, leading to the incorrect merging of `isOptimisticReport` into the old assignee's account.